controllers/stresschaos/types.go

```diff
@@ -0,0 +1,254 @@
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/pingcap/chaos-mesh/api/v1alpha1"
+	"github.com/pingcap/chaos-mesh/controllers/common"
+	"github.com/pingcap/chaos-mesh/controllers/reconciler"
+	"github.com/pingcap/chaos-mesh/controllers/twophase"
+	pb "github.com/pingcap/chaos-mesh/pkg/chaosdaemon/pb"
+	"github.com/pingcap/chaos-mesh/pkg/utils"
+)
+
+const stressChaosMsg = "stressors: %v"
+
+// Reconciler is stresschaos reconciler
+type Reconciler struct {
+	client.Client
+	record.EventRecorder
+	Log logr.Logger
+}
+
+// Reconcile reconciles a StressChaos resource
+func (r *Reconciler) Reconcile(req ctrl.Request, chaos *v1alpha1.StressChaos) (ctrl.Result, error) {
+	r.Log.Info("Reconciling stresschaos")
+	scheduler := chaos.GetScheduler()
+	duration, err := chaos.GetDuration()
+	if err != nil {
+		r.Log.Error(err, fmt.Sprintf("unable to get stresschaos[%s/%s]'s duration", chaos.Namespace, chaos.Name))
+		return ctrl.Result{}, err
+	}
+	if scheduler == nil && duration == nil {
+		return r.commonStressChaos(chaos, req)
+	} else if scheduler != nil && duration != nil {
+		return r.scheduleStressChaos(chaos, req)
+	}
+
+	// This should be ensured by admission webhook in the future
+	r.Log.Error(fmt.Errorf("stresschaos[%s/%s] spec invalid", chaos.Namespace, chaos.Name), "scheduler and duration should be omitted or defined at the same time")
+	return ctrl.Result{}, fmt.Errorf("invalid scheduler and duration")
+}
+
+func (r *Reconciler) commonStressChaos(stresschaos *v1alpha1.StressChaos, req ctrl.Request) (ctrl.Result, error) {
+	cr := common.NewReconciler(r, r.Client, r.Log)
+	return cr.Reconcile(req)
+}
+
+func (r *Reconciler) scheduleStressChaos(stresschaos *v1alpha1.StressChaos, req ctrl.Request) (ctrl.Result, error) {
+	sr := twophase.NewReconciler(r, r.Client, r.Log)
+	return sr.Reconcile(req)
+}
+
+// Apply applies stress-chaos
+func (r *Reconciler) Apply(ctx context.Context, req ctrl.Request, chaos reconciler.InnerObject) error {
+	stresschaos, ok := chaos.(*v1alpha1.StressChaos)
+	if !ok {
+		err := errors.New("chaos is not stresschaos")
+		r.Log.Error(err, "chaos is not StressChaos", "chaos", chaos)
+		return err
+	}
+
+	pods, err := utils.SelectAndFilterPods(ctx, r.Client, &stresschaos.Spec)
+
+	if err != nil {
+		r.Log.Error(err, "failed to select and generate pods")
+		return err
+	}
+
+	stresschaos.Status.Instances = make(map[types.UID]v1alpha1.StressInstance)
+	err = r.applyAllPods(ctx, pods, stresschaos)
+	if err != nil {
+		r.Log.Error(err, "failed to apply chaos on all pods")
+		return err
+	}
+
+	stresschaos.Status.Experiment.Pods = []v1alpha1.PodStatus{}
+
+	for _, pod := range pods {
+		ps := v1alpha1.PodStatus{
+			Namespace: pod.Namespace,
+			Name:      pod.Name,
+			HostIP:    pod.Status.HostIP,
+			PodIP:     pod.Status.PodIP,
+			Message:   fmt.Sprintf(stressChaosMsg, stresschaos.Spec.Stressors),
```

Comment:
`stresschaos.Spec.Stressors` is a pointer, so the result looks like below. Please use a string function here.

``` yaml
  status:
    experiment:
      phase: Running
      podChaos:
      - action: ""
        hostIP: 172.17.0.4
        message: 'stressors: &{<nil> 0xc0001786e0}'
        name: etcd-7db9d6f6f9-lgr6k
        namespace: chaos-testing
        podIP: 10.244.1.8
      startTime: "2020-04-24T20:34:53Z"
```